File: src/GymManagement.Domain/Entities/Warmup.cs
namespace GymManagement.Domain.Entities;

public class Warmup : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? VideoUrl { get; set; }
    public int DurationSeconds { get; set; }
    public string? DifficultyLevel { get; set; }
    public string? BodyPart { get; set; }
    public int? CaloriesBurn { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<WorkoutPlanWarmup> WorkoutPlanWarmups { get; set; } = new List<WorkoutPlanWarmup>();
}

File: gym_client/src/types/warmup.ts
export interface Warmup {
  id: number
  name: string
  description?: string | null
  videoUrl?: string | null
  durationSeconds: number
  difficultyLevel?: string | null
  bodyPart?: string | null
  caloriesBurn?: number | null
  isActive: boolean
}

export interface CreateWarmupDto {
  name: string
  description?: string | null
  videoUrl?: string | null
  durationSeconds: number
  difficultyLevel?: string | null
  bodyPart?: string | null
  caloriesBurn?: number | null
  isActive?: boolean
}

export interface UpdateWarmupDto {
  name?: string
  description?: string | null
  videoUrl?: string | null
  durationSeconds?: number
  difficultyLevel?: string | null
  bodyPart?: string | null
  caloriesBurn?: number | null
  isActive?: boolean
}

export interface PagedWarmups {
  items: Warmup[]
  totalCount: number
  page: number
  pageSize: number
}

export interface WorkoutPlanWarmup {
  id: number
  warmupId: number
  name: string
  description?: string | null
  videoUrl?: string | null
  durationSeconds: number
  difficultyLevel?: string | null
  bodyPart?: string | null
  caloriesBurn?: number | null
  displayOrder: number
}

export interface PlanWarmupWriteDto {
  warmupId: number
  displayOrder: number
}
